cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1763,10 +1763,10 @@ auto convert_table_to_parquet_data(table_input_metadata& table_meta,
     // for multiple fragments per page to smooth things out. using 2 was too
     // unbalanced in final page sizes, so using 4 which seems to be a good
     // compromise at smoothing things out without getting fragment sizes too small.
-    auto frag_size_fn = [&](auto const& col, size_type col_size) {
+    auto frag_size_fn = [&](auto const& col, size_t col_size) {
       int const target_frags_per_page = is_col_fixed_width(col) ? 1 : 4;
       auto const avg_len =
-        target_frags_per_page * util::div_rounding_up_safe<size_type>(col_size, input.num_rows());
+        target_frags_per_page * util::div_rounding_up_safe<size_t>(col_size, input.num_rows());
```

Comment:
It seems that the overflow is due to multiplication here. So probably this instead of changing `col_size` type:
```suggestion
        target_frags_per_page * static_cast<std::size_t>(util::div_rounding_up_safe<size_t>(col_size, input.num_rows()));
```

Reply:
The correct type at L1766 should indeed be `size_t` to begin with and the `size_type` is actually an implicit typecasting.